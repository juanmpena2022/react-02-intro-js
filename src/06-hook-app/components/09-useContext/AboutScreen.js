import { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);
  const logout = () => {
    setUser(null);
  };

  return (
    <>
      <h1>AboutScreen</h1>
      <hr />

      <pre>{JSON.stringify(user, null, 3)}</pre>
      <button className="btn btn-primary" onClick={logout}>Logout</button>
    </>
  );
};
