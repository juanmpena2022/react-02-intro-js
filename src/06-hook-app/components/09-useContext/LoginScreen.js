import { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);
  const loginUser = () => {
    setUser({
      id: 1,
      name: "Juan",
      email: "jm@gmail.com",
    });
  };

  return (
    <>
      <h1>LoginScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={loginUser}>
        Login
      </button>
    </>
  );
};
