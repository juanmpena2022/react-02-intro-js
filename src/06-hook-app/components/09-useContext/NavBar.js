import { Link } from "react-router-dom";

export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <Link className="navbar-brand" to="/">
        useContext
      </Link>
      <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
        <div className="navbar-nav">
          <Link className="nav-item nav-link" to="/">
            Home
          </Link>
          <Link className="nav-item nav-link" to="/about">
            About
          </Link>
          <Link className="nav-item nav-link" to="/login">
            Login
          </Link>
        </div>
      </div>
    </nav>
  );
};
