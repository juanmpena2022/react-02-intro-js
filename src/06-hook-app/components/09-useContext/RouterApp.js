import React from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";

import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./NavBar";

export const RouterApp = () => {
  return (
    <BrowserRouter>
      <NavBar />
      <div className="container">
        <Routes>
          <Route path="/about" element={<AboutScreen />} />
          <Route path="/login" element={<LoginScreen />} />
          <Route path="/" element={<HomeScreen />} />
        </Routes>
      </div>
    </BrowserRouter>
  );
};
