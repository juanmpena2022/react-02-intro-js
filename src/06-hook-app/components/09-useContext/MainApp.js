import { useState } from "react";
import { RouterApp } from "./RouterApp";
import { UserContext } from "./UserContext";

export const MainApp = () => {
  const [user, setUser] = useState(null);
  var userContextProviderValue = { user, setUser };

  return (
    <UserContext.Provider value={userContextProviderValue}>
      <RouterApp />
    </UserContext.Provider>
  );
};
