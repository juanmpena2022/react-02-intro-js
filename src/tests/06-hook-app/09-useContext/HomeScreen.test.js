import { shallow, mount } from "enzyme";
import { HomeScreen } from "../../../06-hook-app/components/09-useContext/HomeScreen";
import { UserContext } from "../../../06-hook-app/components/09-useContext/UserContext";

describe("pruebas del componente <HomeScreen />", () => {
  const user = {
    name: "Juan",
    email: "jm@gmail.com",
  };

  test("snapshot", () => {
    const wrapper = mount(
      <UserContext.Provider value={{ user }}>
        <HomeScreen />
      </UserContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
