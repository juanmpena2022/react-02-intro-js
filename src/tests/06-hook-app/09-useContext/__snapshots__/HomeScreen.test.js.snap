// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas del componente <HomeScreen /> snapshot 1`] = `
Array [
  <h1>
    HomeScreen
  </h1>,
  <hr />,
  <pre>
    {
   "name": "Juan",
   "email": "jm@gmail.com"
}
  </pre>,
]
`;
